### PR TITLE
chore(deps): Updates gradle version to 8.2.2.  Updates kotlin to 1.9.22.

### PR DESCRIPTION
**Pull request by dkhawk**
The current Pull Request introduces a general update of dependencies, and supersedes all the existing dependabot PRs.

This PR affecs the demo-java, demo-kotlin and snippets projects.

Increased compileSdkVersion.
Increased targetSdkVersion.
Increases Gradle plugin version.
General dependencies updated.

---
name: Pull request
about: Create a pull request
label: 'triage me'
---
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open a GitHub issue as a bug/feature request before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: demo-java/app/build.gradle [hunk 1]
```diff
@@ -29,30 +29,31 @@ android {
...
     // To add your Google Maps Platform API key to this project:
-    // 1. Create or open file local.properties in this folder, which will be read by default
-    //    by secrets_gradle_plugin
-    // 2. Add this line, replacing YOUR_API_KEY with a key from a project with Places API enabled:
-    //        PLACES_API_KEY=YOUR_API_KEY
-    // 3. Add this line, replacing YOUR_API_KEY with a key from a project with Maps SDK for Android
-    //    enabled (can be the same project and key as in Step 2):
-    //        MAPS_API_KEY=YOUR_API_KEY
+    // 1. Copy local.defaults.properties to secrets.properties
+    // 2. Replace PLACES_API_KEY=DEFAULT_API_KEY with a key from a project with Places API enabled
```
Comment (wangela) [inline demo-java/app/build.gradle:51]: For steps 2 and 3, "In secrets.properties, ..." to make it clear which file to edit.

File: demo-kotlin/app/build.gradle [hunk 2]
```diff
@@ -48,13 +48,13 @@ repositories {
...
-    // 1. Create or open file secrets.properties in the root folder of the project, which will be
-    // read by secrets_gradle_plugin
-    // 2. Add this line, replacing YOUR_API_KEY with a key from a project with Places API enabled:
-    //        PLACES_API_KEY=YOUR_API_KEY
-    // 3. Add this line, replacing YOUR_API_KEY with a key from a project with Maps SDK for Android
-    //    enabled (can be the same project and key as in Step 2):
-    //        MAPS_API_KEY=YOUR_API_KEY
-    propertiesFileName = "secrets.properties"
+    // 1. Copy local.defaults.properties to secrets.properties
+    // 2. Replace PLACES_API_KEY=DEFAULT_API_KEY with a key from a project with Places API enabled
```
Comment (wangela) [inline demo-kotlin/app/build.gradle:52]: Same edit

File: snippets/gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.0-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.4-all.zip
```
Comment (wangela) [inline snippets/gradle/wrapper/gradle-wrapper.properties:6]: Why is this one upgraded to 8.4 whereas the others are at 8.2?

**Review by wangela — CHANGES_REQUESTED** [review]

File: snippets/gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
[the remark this answers is not in the document]
Comment (dkhawk) [reply snippets/gradle/wrapper/gradle-wrapper.properties:6]: I used Android Studio.  Updating to 8.4 was not intended.  Thanks for catching this.

File: demo-kotlin/app/build.gradle (hunk 2 above)
Comment (dkhawk) [reply demo-kotlin/app/build.gradle]: Done

File: snippets/gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (dkhawk) [reply snippets/gradle/wrapper/gradle-wrapper.properties]: Studio was too aggressive.  :-)

**Review by wangela — APPROVED** [review]

**Pull request merged**